src/main/java/edu/umich/its/spe/SPEMaster.java

```diff
@@ -0,0 +1,378 @@
+		
+		if (!verifyESB()) {
+			M_log.error("Unable to connect to esb"); 
+			throw new SPEEsbException("Unable to connect to ESB");
+		}
+		
+		//// Get the relevant grades.
+		String lastUpdateTime = readLastGradeTransferTime();
+		M_log.error("fake lastUpdateTime");
+		lastUpdateTime = "2017-03-08 15:38:21";
+		M_log.info("lastUpdateTime: {}",lastUpdateTime);
+		String assignmentsFromDW;
+	
+		try {
+			assignmentsFromDW = getSPEGrades(lastUpdateTime);
+	
+			//// Extract grades from JSON the grades for insertion	
+	
+			ArrayList<HashMap<String,String>> SPEgradeMaps = null;
+			try {
+				SPEgradeMaps = convertSPEGradesFromDataWarehouseJSON(assignmentsFromDW);
+			} catch (JSONException e) {
+				M_log.error("exception in converting grades: "+e);
+			}
+	
+			//// Insert the grades.
+			M_log.info("Grade count since {} is {}.",lastUpdateTime,SPEgradeMaps.size());
+			putSPEGrades(SPEgradeMaps);
+	
+		} catch (SPEEsbException e1) {
+			M_log.error("Exception processing SPE grades:",e1);
+		}
+		finally {
+			// Finish up, save data, send reports.
+			closeUpShop();
+		}
+	}
+
+	/************ Close down processing, create summary ************/
+	public void closeUpShop() {
+		// TODO: implement task shutdown.
+		M_log.error("Implement close up shop");
+	}
+
+	/********* esb verify **********/
+	// Run sanity check to make sure can talk to ESB and get data.
+	
+	/*
+	 * Setup the hash of values needed for the call to verify ESB connection.
+	 */
+	protected HashMap<String, String> setupESBVerifyCall() {
+		//M_log.debug("spe properties: "+speproperties);
+		HashMap<String,String> value = new HashMap<String,String>();
+		value.putAll(speproperties.getEsb());
+		return value;
+	}
+
+	public Boolean verifyESB() {
+		// TODO: implement verify by renewing access token.
+		M_log.error("verify is not yet used by SPEMaster");
+		HashMap<String,String> value = setupESBVerifyCall();
+		return speesb.verify(value);	
+	}
+
+	/*************** get grades via ESB **************/
+
+	/*
+	 * Setup the hash of values needed for the call to get grades.
+	 */
+	protected HashMap<String, String> setupGetGradesCall(String gradedAfterTime) {
+		M_log.debug("spe properties: "+speproperties);
+		HashMap<String,String> value = new HashMap<String,String>();
+		value.putAll(speproperties.getEsb());
+		value.putAll(speproperties.getGetgrades());
+		value.put("gradedaftertime", gradedAfterTime);
+		return value;
+	}
+
+	/*
+	 * Get grades as JSON string.  Only grades after the gradeAfterTime
+	 * timestamp will be returned. An empty result is reasonable.
+	 * The format of the time stamp is: 2017-04-01 18:00:00.  
+	 * TODO: timestamp default value?
+	 */ 
+
+	public String getSPEGrades(String gradedAfterTime) throws SPEEsbException {
+
+		HashMap<String, String> values = setupGetGradesCall(gradedAfterTime);
+
+		WAPIResultWrapper grades = speesb.getGradesViaESB(values);
+		// check for possibility of no new grades.
+		if (grades.getStatus() == 404) {
+			return "[]";
```

Comment:
There will be other ways to track the number so I don't think it needs a special message.  This also isn't an error, just something that can happen (frequently).